osm2gtfs/creators/stops_creator.py

```diff
@@ -18,85 +15,99 @@ def __repr__(self):
+        It can be overridden by custom creators to change how stop_ids are made
+        up.
+
+        :return stop_id: A string with the stop_id in the GTFS
+        """
+        try:
+            parent_station = stop.get_parent_station()
+        except AttributeError:
+            parent_station = ""
+
+        # Send stop_id creation through overridable function
+        gtfs_stop_id = self._define_stop_id(stop)
+
+        # Save defined stop_id to the object for further use in other creators
+        stop.set_stop_id(gtfs_stop_id)
+
+        # Set stop name
+        stop_name = self._define_stop_name(stop)
+
+        # Collect all data together for the stop creation
+        field_dict = {'stop_id': self._define_stop_id(stop),
+                      'stop_name': stop_name,
+                      'stop_lat': float(stop.lat),
+                      'stop_lon': float(stop.lon),
+                      'location_type': stop.location_type,
+                      'parent_station': parent_station
+                      }
 
         # Add stop to GTFS object
-        stop = transitfeed.Stop(field_dict=stop_dict)
-        feed.AddStopObject(stop)
-        return stop
+        feed.AddStopObject(transitfeed.Stop(field_dict=field_dict))
 
-    def add_stops_to_routes(self, data):
-        routes = data.routes
-        stops = data.stops
+        if type(stop_name).__name__ == "unicode":
+            stop_name = stop_name.encode('utf-8')
 
-        # Loop through routes
-        for ref, route in routes.iteritems():
-            # Replace stop ids with Stop objects
-            self._fill_stops(stops, route)
+        print("Added stop: " + stop_name +
+              " - " + stop.osm_url)
 
-        data.routes = routes
-        return
+        # Return the stop_id of the stop added
+        return field_dict['stop_id']
 
-    def _fill_stops(self, stops, route):
+    def _define_stop_id(self, stop):
         """
-        Fill a route object with stop objects for of linked stop ids
+        This function returns the GTFS stop id to be used for a stop.
+        It can be overridden by custom creators to change how stop_ids are made
+        up.
+
+        :return stop_id: A string with the stop_id for use in the GTFS
         """
-        if isinstance(route, Route):
-            i = 0
-            for stop in route.stops:
-                # Replace stop id with Stop objects
-                # TODO: Remove here and use references in TripsCreatorFenix
-                route.stops[i] = self._get_stop(stop, stops)
-                i += 1
-
-        elif isinstance(route, RouteMaster):
-            for route_variant_ref, route_variant in route.routes.iteritems():
-                self._fill_stops(stops, route_variant)
 
+        #  Use a GTFS stop_id coming from OpenStreetMap data
+        if "ref:gtfs" in stop.tags:
+            stop_id = stop.tags['ref:gtfs']
+        elif "ref" in stop.tags:
+            stop_id = stop.tags['ref']
```

Comment:
Using something from OpenStreetMap here might get you into trouble when the tags there change later on.

Reply:
Yes, indeed, will move this check to the osm specific code in `osm_connector.py`.

Reply:
I do think we should use in the standard stops_creator the tags from OSM this is important being able to create connection between existing, official GTFS to the data in OpenStreetMap and also relevant for the idea of using WikiData for linking.